libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/NativeLoginManager.kt

```diff
@@ -123,13 +124,8 @@ internal class NativeLoginManager(
         val trimmedUsername = username.trim()
         val trimmedPassword = password.trim()
 
-        if (!isValidUsername(trimmedUsername)) {
-            return InvalidUsername
-        }
-
-        if (!isValidPassword(trimmedPassword)) {
-            return InvalidPassword
-        }
+        trimmedUsername.mapInvalidUsernameToResult()?.let { return it }
```

Comment:
These parameter validations are used in a lot of places now, so I made them into a one-liner.  The original method is still used and tested, so this is just optional syntactic sugar.